### ISSUE TITLE
[Funding]: Develop a spending plan

### ISSUE THREAD

**Issue by fricklerhandwerk** [NixOS/foundation#129]
# Description

The NGI0 grant agreements provide the foundation with financial means to sustain its operation. Concretely this amounts to ca. 35 000 EUR just this year, and a total of ca. 120 000 EUR until end of 2026. This money could be invested in maintaining and developing critical code that helps keep the ecosystem afloat.

Judging from experience, the board does not have the time to make spending decisions at that granularity, which would also often involve technical decisions; not even speaking of seeing projects through to the end.

My proposal is to gather projects to be funded from that budget, including people to implement them, and present the board with a big-picture recommendation to sign off. This would happen in a similar manner as it did with the [STF proposal in 2023](https://discourse.nixos.org/t/german-federal-funding-for-foss-development/29036/4), probably even more transparent because less rushed.

Technically the budget is already in scope of my current responsibilities under the NGI project manager contract, so in principle I could just go ahead and do it. But I'd like to have a clear stance from the board on that issue, since it's a strategic one, and ideally also feedback from the contributor community.

# Budget amount

3 000 EUR (from the NGI0 grants)

# Budget description

Paid time for me to do the work. We can specify a deadline, a format of the proposal, and other boundary conditions, if the general direction is agreeable.

**Comment by zimbatm** [comment]
If we are starting to receive more grants, it would be good to have a central place where people can make propositions for projects and check which grants are available. And then have the Nix community team members help pick the candidates. Until then, I think your proposal makes sense to me.

From the board's perspective, it keeps us away from making technical decisions when picking which project is worthwhile or not. On more practical terms, we have to be able to know which project maps to which funds when receiving the invoices.

**Comment by Janik-Haag** [comment]
> The NGI0 grant agreements provide the foundation with financial means to sustain its operation. Concretely this amounts to ca. 35 000 EUR just this year, and a total of ca. 120 000 EUR until end of 2026. This money could be invested in maintaining and developing critical code that helps keep the ecosystem afloat.

This sounds very reasonable to me, there are multiple areas in the project that are figuratively speaking on fire and reaching from just a small dumpster fire to forest fire.
Aiding them with funding can be a good way to mitigate those issues.

> Judging from experience, the board does not have the time to make spending decisions at that granularity, which would also often involve technical decisions; not even speaking of seeing projects through to the end.

Can you elaborate on the granularity? As far as I know the board just barely gets Project proposals so having an official process for submitting projects would be nice.


> My proposal is to gather projects to be funded from that budget, including people to implement them, and present the board with a big-picture recommendation to sign off. This would happen in a similar manner as it did with the STF proposal in 2023, probably even more transparent because less rushed.

Do you have anything in particular in mind for your "big-picture"?

Why would you want to gather the projects and people to do so if we could work on a self-sustaining process that shift's the responsibility of collecting this information to the teams/people interested in getting funding. Another organization where this appears to work well would be the [FreeBSD Foundation](https://freebsdfoundation.org/our-work/projects/)
We can ask teams to rank their proposal by importance and then have the foundation make the call.

> Technically the budget is already in scope of my current responsibilities under the NGI project manager contract, so in principle I could just go ahead and do it. But I'd like to have a clear stance from the board on that issue, since it's a strategic one, and ideally also feedback from the contributor community.

If you actually want feedback from the community you have to go and get it, just opening an issue in the foundation repo probably won't do anything.
As I already told you with the proposal for #37 if you want involvement, just do a discourse post looking for interested people.

> PS: This effort could also involve planning or already doing some fundraising.

On the topic of fundraising we should try to take some inspirations from other Projects like [Debian](https://wiki.debian.org/Fundraising).

CC @ronef since you already had some ideas for fundraising and have it on your priority list.
CC @arianvp since you offered some help on that topic in the past

> PPS: There is of course a short-cut where we simply divide the money across community teams as proposed in NixOS/teams-collaboration#2

No need for taking short-cut's here since we have enough time to take a cautions decision.

---

Thinking about this a bit a few Projects that would be interesting to fund:

### Hydra frontend

The infra team expressed their interest to rewrite the hydra frontend in Django cleaning up old Perl code and making it easier to maintain.
Some discussion on this can be found [on matrix](https://matrix.to/#/!RROtHmAaQIkiJzJZZE:nixos.org/$jKncoLvRideSkPCH-3brXK_jTcsci-GrC2GqwhQGplI?via=nixos.org&via=matrix.org&via=nixos.dev) in #infra:nixos.org

cc @delroth since you initiated the discussion and are part of the infra team.

### Nix source tree modification tool

In nixpkgs doing tree wide changes is a huge pain and doesn't get done as often as it should.
I have recently learned about [nixq](https://stuebinm.eu/git/nixq/tree/) but am unfamiliar how far the development process on it is and what might be potential pain points.
It was already used for two nixpkgs prs. [1](https://github.com/NixOS/nixpkgs/pull/297084) [2](https://github.com/NixOS/nixpkgs/pull/303841)

cc @stuebinm since you are the author of nixq
cc @teto since you expressed interest in writing such a tool a while ago in #dev:nixos.org

### Ofborg improvements

Ofborg has sever testability issues and some other things.
IIRC it was already discussed as a project for STF but since round two didn't happen the project didn't happen either.
I assume there already is a fairly concrete proposal for this because of the STF history.

cc @lilyinstarlight

### Mod team Code of Conduct (CoC) training

The moderation team has been struggling a lot with member retention and burnout.
This is in part due to uncertainty of how to enforce our Code of Conduct and has become more apparent in the light of the sponsorship discussion.
An example for such a training would be: https://otter.technology/code-of-conduct-training/
Note that this might also be an interesting coaching for the foundation board itself.

### Hardware for releases.nixos.org and ceph hosting

In an effort to reduce the S3 AWS cost it was talked about self-hosting cache.nixos.org
There already have been a bunch of discussion around self-hosting and releases.nixos.org would be a good testbed while reducing AWS bandwidth cost.

### nixpkgs observability tools

In nixpkgs currently we have the issue of people often repeating work already done by others, e.g. packaging the same thing multiple times.
Tools for finding issues and PRs and being able to query them directly would be very helpful here since GitHub's search already isn't the best.
And the current search experience will only become worse once we have a merge bot and can't see who merged what which could also be mitigated using a tool like this.

**Comment by fricklerhandwerk** [comment]
After discussing the matter with various people I suggest we use the operational budget for the following:

> tl;dr Use the funds for 3 paid part-time contributors in admin, infra, and outreach. I offer to do the supervision.

- Improving accounting, budgeting, and reporting

  [The current setup](https://github.com/NixOS/foundation/issues/163) is, in my humble opinion, quite the mess. Some of that can't go away and we just have to deal with it, but it's hard to keep track of things. We're wasting substantial human time on manual reconciliation of accounting information, part of which is due to an organically grown information architecture. This can be fixed with additional focus time by the foundation's admin assistant, who would work together with the board, the SC, and the teams.

- Building organisation-level infrastructure to be able to better handle ongoing and future projects

  The other part of messiness is due to our tooling being limited in various ways. This can be fixed by
  - Setting up open source tools for various purposes

     In particular that would be for accounting, but also video calls, calendaring, private notes, etc. After years, we still rely on @Lassulus generous support allowing us to use his Jitsi and Hedgedoc servers.
  - Improving automation to aggregate or synchronise accounting data from various sources.

    It's already evident that we would benefit a lot from a server to run periodic tasks against a couple of interfaces, or calling web hooks in existing platforms. Many teams could benefit from the general availability of such infrastructure that they could hook into or build upon. I can imagine e.g. the marketing team to benefit from some automation with regards to outreach activities and data aggregation.

  Concretely, and I've discussed this with @nixos/infra for the past half year, I suggest funding time for @erethon to support the infra team as an on-call devops to triage issues, review PRs to https:/github.com/nixos/infra, and work with community teams to make sure they have all the tools they need.

- Fundraising and partnerships

  On request of @tomberek from the SC, Kate @idabzo and me started hashing out how to improve our handling of partnerships (such as with Framework, GitHub, Google Summer of Code, etc. -- a lot of that stuff flies under the radar or tends to slip through the cracks if there's no paper trail) and improve effectiveness of fundraising, e.g. via [Open Collective](https://opencollective.com/nixos). The number of opportunities to get in-kind donations or other tangible benefits through partnerships is rising steadily, but someone needs to handle those contacts, follow up timely, communicate requirements to teams and results to the public. All this requires availability and time, and funding that would help improve our ability e.g. to keep operating mission-critical infrastructure.

- Outreach

  In order to demonstrate that we're actually doing something -- oftentimes amazing work is simply not particularly visible -- we could put more emphasis on structuring and streamlining our broadcast communication. The marketing team already agreed to have [a dedicated team lead](https://discourse.nixos.org/t/2025-01-13-marketing-team-minutes/58768) with @idabzo, and funding regular activities would make a huge difference in the team's effectiveness and quality of life for volunteers. There's also useful work that could be done to keep improving the visual presentation @avocadoom has been working on in the past year.
 
  Both would help with fundraising and partnership activities, if we could more easily point at evidence of past success stories. 

@NixOS/steering @NixOS/foundation what do we need to get an approval?

**Comment by fricklerhandwerk** [comment]
Small update. After working through the books with @kpark955 and @ryantrinkle to clarify the foundation's financial situation and draft a forecast for 2025, yesterday at #ZurichZHF with @lassulus we determined that the next step will be to address this issue (finally!) and determine priorities on what to use the available resources for. Very likely fundraising in the broadest sense.
